src/intervals/functions.jl

```diff
@@ -373,3 +373,18 @@ function nthroot(a::Interval{T}, n::Integer) where T
     b = nthroot(bigequiv(a), n)
     return convert(Interval{T}, b)
 end
+
+"""
+Calculate `x::Interval mod y::Real`, limited by `y != 0`.
+"""
+function mod(x::Interval, y::Real)
+    @assert y != zero(y) """mod(x::Interval, y::Real) 
```

Comment:
an alternative would be to return an empty interval, I think this would be more conformant with the IEEE standard behavior, e.g.

```julia
julia> sqrt(-1.. -1)
∅

julia> log(-2.. -1)
∅
```

Reply:
Or we can define `mod(a..b, 0) == 0..0 ⊇ ∅` as Wolfram says
https://www.wolframalpha.com/input?i=limit+x-%3E0%2B+mod%281%2C+x%29

Consequently, for `0 ∈ y`, we can define smth like `mod(x,y) = union(mod(x,(y.lo..y.lo/3)), mod(x, (y.hi/3)..(y.hi)), 0..0)` because the maximum value needs to by in `(y.hi/2)..(y.hi)`

Reply:
I propose to solve this in some future PR, and move the discussion to #129, to keep track once this PR merged.

Reply:
> Or we can define `mod(a..b, 0) == 0..0 ⊇ ∅` as Wolfram says https://www.wolframalpha.com/input?i=limit+x-%3E0%2B+mod%281%2C+x%29

Julia (and Wolfram) return `NaN` (`undefined`) for `mod(3.2, 0.0)`, which makes sense due to the division in the algorithm.

> 
> Consequently, for `0 ∈ y`, we can define smth like `mod(x,y) = union(mod(x,(y.lo..y.lo/3)), mod(x, (y.hi/3)..(y.hi)))` because the maximum value needs to by in `(y.hi/2)..(y.hi)`

This idea is interesting, and reminds me a little bit what we have for `extended_div`. That certainly should be addressed in another PR. Yet, I think for (mathematical) reals, `mod(x,y)` is a number in the interval [0,y] for y>0, or [y, 0] if y<0, isn't it?

From the range of `mod(x,y)` for real `y`, then for a strictly positive interval `y`, we should return `[0, sup(y)]`, or if it is strictly negative `[inf(y), 0]`. If `0 ∈ y`, we should return the hull of these intervals.

Reply:
I think my last comment corresponds to [these results](https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/525#discussion_r883635954)... Or not?